Libraries/src/net/relinc/libraries/sample/CompressionSample.java

```diff
@@ -20,9 +21,16 @@ public void setHoppySpecificParameters(String des, String val) {
 	}
 
 	@Override
-	public String getHoppySpecificString() {
-		return getDiameter() > 0 ? "Diameter" + delimiter + getDiameter() + SPSettings.lineSeperator : "";
+	public void setHoppySpecificParametersJSON(JSONObject jsonObject) {
+		setDiameter((Double)jsonObject.get("Diameter"));
 	}
+
+
+	@Override
+	public void getHoppyAddSpecific(JSONObject jsonObject) {
```

Comment:
Side-effect programming. Prefer not to write code like this. Prefer to return a new `JSONObject`, then have the caller merge the new data. 
If possible, it's preferred to have methods that do not rely on any outside state except the arguments, and prefer not to change any outside state (which is what you're doing here). Prefer to return the value to be consumed by the caller.